File: 287.py
class Solution:
    def findDuplicate(self, nums: list[int]) -> int:
        s = 0
        f = 0
        while True:
            f = nums[nums[f]]
            s = nums[s]
            if f == s:
                break
        s2 = 0
        while s != s2:
            s = nums[s]
            s2 = nums[s2]
        return s

File: 287.cpp
#include <vector>
using namespace std;

class Solution
{
  public:
    int findDuplicate(vector<int> &nums)
    {
        int s = 0, f = 0, s2 = 0;
        do
        {
            s = nums[s];
            f = nums[nums[f]];
        } while (s != f);
        while (s != s2)
        {
            s = nums[s];
            s2 = nums[s2];
        }
        return s;
    }
};